guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/smallest-common-multiple/index.md

```diff
@@ -187,6 +187,46 @@ Note: If the array only has two elements, then the `for` loop never gets used an
 
 *   <a href='https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Increment' target='_blank' rel='nofollow'>Prefix increment operator ++</a>
 
+## ![:rotating_light:](https://forum.freecodecamp.com/images/emoji/emoji_one/rotating_light.png?v=3 ":rotating_light:") Alternate Advanced Code Solution:
+```
+const smallestCommons = arr => {
+  const MIN = Math.min(...arr);
+  const MAX = Math.max(...arr);
+  
+  // Fill a new array with values as min increments by one to max
+  const smallestCommon = Array(MAX - MIN + 1)
+    .fill()
+    .map((item, index) => MIN + index)
+    // Reduce to the smallest common multiple
+    .reduce(lcm);
+
+  return smallestCommon;
+};
+
+/* Determine the smallest common multiple by repeatedly adding the 
+*  current value until it is divisible by the sum
+*/
+const lcm = (sum, curr) => {
+  let common = curr;
+  while (common % sum !== 0) {
+    common += curr;
+  }
+  return common;
+};
+    
+smallestCommons([1,5]);
+```    
+### Code Explanation:
+
+*   Create a new array filled with all numbers between the starting min and max.
+*   Immediately determine the LCM by reducing the array.
+*   Skipping the loops and sorting needed to calculate the GCD first is significantly more efficient in Firefox 65 (up to 9.69%). However, in Chrome it is marginal efficiency (+/- 1%).
```

Comment:
```suggestion
* *Note:*  Skipping the loops and sorting needed to calculate the GCD first is significantly more efficient in Firefox 65 (up to 9.69%). However, in Chrome it is marginal efficiency (+/- 1%).
```
I think it is better as a note rather than a code explanation
cc: @ojeytonwilliams @RandellDawson

Reply:
Yours looks better

Reply:
Just checking - Should I commit the suggestion or wait for Randell Dawson?

Reply:
It's fine to commit it.  @RandellDawson will get notified of any commits you add to this and can make further suggestions if he wants to change something.

Reply:
Thank ya kindly!

Reply:
No problem.  Just to clarify, I think we should give Randell a bit of time to review this, if he wants.  It was just that the commit might as well go in, regardless.

Reply:
I understand and appreciate the clarification, I'm trying to get a feel for how conversations/processes flow.